molgenis-api-permissions/src/main/java/org/molgenis/api/permissions/model/response/LabelledPermissionResponse.java

```diff
@@ -0,0 +1,49 @@
+package org.molgenis.api.permissions.model.response;
+
+import com.google.auto.value.AutoValue;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.molgenis.util.AutoGson;
+
+@AutoValue
+@AutoGson(autoValueClass = AutoValue_LabelledPermissionResponse.class)
+@SuppressWarnings(
+    "squid:S1610") // Abstract classes without fields should be converted to interfaces
+public abstract class LabelledPermissionResponse {
```

Comment:
Is it correct that all field can be null? I would expect at least one non-null value.

Reply:
caused by the fact that either the user or role is present, I'll add a check in the create method